pokedex/src/components/Detail.js

```diff
@@ -1,176 +1,77 @@
-                    <td>
-                      <p className="statBar">
-                        <span className="stat" style={{ paddingRight: getStyle(speed) }}>
-                          {speed}
-                        </span>
-                      </p>
-                    </td>
-                  </tr>
-                  <tr>
-                    <td>Special Attack</td>
-                    <td>
-                      <p className="statBar">
-                        <span className="stat" style={{ paddingRight: getStyle(specialAttack) }}>
-                          {specialAttack}
-                        </span>
-                      </p>
-                    </td>
-                  </tr>
-                  <tr>
-                    <td>Special Defense</td>
-                    <td>
-                      <p className="statBar">
-                        <span className="stat" style={{ paddingRight: getStyle(specialDefense) }}>
-                          {specialDefense}
-                        </span>
-                      </p>
-                    </td>
-                  </tr>
+                  {Object.entries(stats).map((stat) => (
+                    <StatRow stat={stat[0]} value={stat[1]} key={stat} />
+                  ))}
                 </tbody>
               </table>
             </div>
           </div>
           <h4>{genus}</h4>
           <p>{description}</p>
           <h4 className="profile">Profile</h4>
-          <table className="profileTable">
-            <colgroup>
-              <col span={"1"} style={{ width: "15%" }} />
-              <col span={"1"} style={{ width: "30%" }} />
-              <col span={"1"} style={{ width: "20%" }} />
-              <col span={"1"} style={{ width: "35%" }} />
-            </colgroup>
-            <tbody>
-              <tr>
-                <th>Height:</th>
-                <td>{height} m</td>
-                <th>Abiblities:</th>
-                <td>
-                  {Object.keys(abilities).map((key) => {
-                    let comma = ",";
-                    if (key == abilities.length - 1) {
-                      comma = "";
-                    }
-                    return <span key={key}>{abilities[key] + comma} </span>;
-                  })}
-                </td>
-              </tr>
-              <tr>
-                <th>Weight:</th>
-                <td>{weight} kg</td>
-                <th>Egg Groups:</th>
-                <td>
-                  {Object.keys(egg_groups).map((key) => {
-                    let comma = ",";
-                    if (key == egg_groups.length - 1) {
-                      comma = "";
-                    }
-                    return <span key={key}>{egg_groups[key] + comma} </span>;
-                  })}
-                </td>
-              </tr>
-            </tbody>
-          </table>
+          <ProfileTable height={height} weight={weight} abilities={abilities} egg_groups={egg_groups} />
```

Comment:
Love it! It's much easier to read not that it's broken off into a smaller component.